ComSemApp/student/views.py

```diff
@@ -928,3 +928,34 @@ def form_valid(self, form):
 
         attempt.save()
         return JsonResponse({'id' : attempt.id}, status=201)
+    
+
+class SpeakingPracticeTeacherReviewRequestCreateView(StudentCourseViewMixin, CreateView):
+    """
+        Used to process and save review request form in SpeakingPracticeResultsView. 
+        Implements the standard Django CreateView
+    """
+
+    model = SpeakingPracticeAttemptReviewRequest
+    fields = []
```

Comment:
Fields should contain `'attempt'`. This will automatically enforce presence and (I believe) the existence of the object referenced by the foreign key.

Reply:
how does this work if I am passing an attempt_id and not the actual object? I tried to do it with attempt in the fields when I was writing this and couldn't get it to work

Reply:
It takes an id, you just have to name it identically to the model field. So, rather than `attempt_id`, you need to send `attempt` in the request.